src/components/__tests__/ConfirmationModal.spec.tsx

```diff
@@ -1,45 +1,43 @@
-import { shallow } from 'enzyme'
+import { fireEvent, render } from '@testing-library/react-native'
 import React from 'react'
-import { View } from 'react-native'
 
+import wrapWithTheme from '../../testing/wrapWithTheme'
 import ConfirmationModal, { ConfirmationModalPropsType } from '../ConfirmationModal'
 
 describe('Components', () => {
   describe('ConfirmationModal ', () => {
+    const setVisible = jest.fn()
+    const confirmationAction = jest.fn()
+
     const defaultModalProps: ConfirmationModalPropsType = {
       visible: false,
-      setVisible: (input: boolean) => {},
+      setVisible: setVisible,
       text: 'Are you sure?',
       confirmationButtonText: 'confirm',
       cancelButtonText: 'cancel',
-      confirmationAction: () => {}
+      confirmationAction: confirmationAction
     }
 
-    it('should have visible property passed to it as default', () => {
-      const modalProps: ConfirmationModalPropsType = {
-        ...defaultModalProps
-      }
-
-      const component = shallow(
-        <View>
-          <ConfirmationModal {...modalProps} />
-        </View>
-      )
-      expect(component.children().props().visible).toBe(false)
+    it('should display passed props', () => {
+      const { getByText } = render(<ConfirmationModal {...defaultModalProps} />, { wrapper: wrapWithTheme })
+      expect(getByText('Are you sure?')).toBeDefined()
+      expect(getByText('cancel')).toBeDefined()
+      expect(getByText('confirm')).toBeDefined()
     })
 
-    it('should have visible property passed to it', () => {
-      const modalProps: ConfirmationModalPropsType = {
-        ...defaultModalProps,
-        visible: true
-      }
+    it('should close on cancel button click', () => {
+      const { getByText } = render(<ConfirmationModal {...defaultModalProps} />, { wrapper: wrapWithTheme })
+      const cancelButton = getByText('cancel')
+      fireEvent.press(cancelButton)
+      expect(setVisible).toBeCalledWith(false)
+      expect(confirmationAction).not.toBeCalled()
```

Comment:
I think you could assert that `navigation.navigate` was called as well with the right params.

Reply:
Navigation is not called anymore. The `ConfirmationModal` is now more generic and can be used for different purposes, e.g. also the confirmation when deleting a module, where no navigation is needed. Therefore the confirmation-action is now passed as a prop and can (but needn't) trigger a navigation